### PR TITLE
Expose faucet config warning through API and Varz

### PR DESCRIPTION
**Pull request by didovesei**
Example of varz:
faucet_config_warning_count 2.0
faucet_config_warning{key="faucet_dp_mac",message="faucet_dp_mac for DPs are not identical"} 1.0
faucet_config_warning{key="nz-kiwi-t1sw2",message="faucet_dp_mac not defined"} 1.0

### CODE REVIEW

File: forch/forchestrator.py [hunk 1]
```diff
@@ -687,8 +694,12 @@ def get_sys_config(self, path, params):
         """Get overall config from faucet config file"""
         try:
             _, _, behavioral_config = self._get_faucet_config()
+            behavioral_config_map = {
+                'content': behavioral_config,
+                'warnings': dict(self._config_summary.warnings)
+            }
             reply = {
-                'faucet_behavioral': behavioral_config,
+                'faucet_behavioral': behavioral_config_map,
```
Comment (grafnu) [inline forch/forchestrator.py:702]: This doesn't seem right and isn't logically consistent.  The warnings are not part of the behavioral config. Seems like the top-level should be faucet (parallel to forch) and then 'content' would be renamed to 'behavorial'.

File: forch/forchestrator.py [hunk 2]
```diff
@@ -628,6 +629,12 @@ def _validate_config(self, config):
                 warnings.append(('faucet_dp_mac', 'faucet_dp_mac for DPs are not identical'))
         return warnings
 
+    def _update_config_warning_varz(self):
+        self._metrics.update_var(
+            'faucet_config_warning_count', len(self._config_summary.warnings))
+        for warning_key, warning_msg in self._config_summary.warnings.items():
+            self._metrics.update_var('faucet_config_warning', 0, [warning_key, warning_msg])
```
Comment (grafnu) [inline forch/forchestrator.py:636]: should be 1 not 0, yes, there's no case where it's anything, but a 0 value implies "this is not true" which is not the case.

File: testing/test_stack [hunk 3]
```diff
@@ -66,6 +66,8 @@ curl http://$CONTROLLER_NAME:9019/protos.html > $OUT_DIR/protos.html
 fgrep "forch/proto/system_state.proto" $OUT_DIR/protos.html || exit 1
 curl http://$CONTROLLER_NAME:9019/sys_config > $OUT_DIR/sys_config.json
 jq '.faucet_behavioral.dps."nz-kiwi-t2sw3"'.faucet_dp_mac $OUT_DIR/sys_config.json | tee -a $TEST_RESULTS
+jq '.faucet_behavioral.warnings.faucet_dp_mac' $OUT_DIR/sys_config.json | tee -a $TEST_RESULTS
+test_varz_value faucet_config_warning_count 8302 forch
```
[the remark this answers is not in the document]
Comment (didovesei) [reply testing/test_stack:70]: yes, just added in etc/test_stack.out

File: forch/forchestrator.py (hunk 1 above)
Comment (didovesei) [reply forch/forchestrator.py]: updated the output structure. PTAL

File: forch/forchestrator.py (hunk 2 above)
Comment (didovesei) [reply forch/forchestrator.py]: Makes sense. Changed to 1

File: forch/forchestrator.py (hunk 1 above)
Comment (didovesei) [reply forch/forchestrator.py]: Now it looks like:

- faucet
   * behavioral
   * structural
   * warnings
- forch
   * ...

**Pull request merged**